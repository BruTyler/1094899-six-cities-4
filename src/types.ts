import {PlaceType} from './const';

export interface WithId<K=string | number> {
  id: K,
}

interface WithLocation {
  location: [number, number]
};

interface DifferentOptions {
  [key: string]: boolean | string | number
};

interface Host {
  avatar: string,
  name: string,
  isSuper: boolean,
}

interface SingleApartmentCard {
  cityId: string,
  type: PlaceType,
  description: string,
  rating: number,
  price: number,
  isPremium: boolean,
  isFavourite: boolean,
  photo: string,
  fullDescription: string,
  photoSet: string[],
  bedrooms: number,
  adultsMax: number,
  goods?: string[],
  host: Host,
}

export type SingleApartment = WithId<number> & WithLocation & SingleApartmentCard;

export type City = WithId<string> & WithLocation & { 
  defaultZoom: number 
};

export type AuthInfo = WithId<number> & DifferentOptions & {
  email: string,
};

export interface AuthData {
  login: string,
  password: string,
};

export type Review = WithId<number> & {
  authorName: string,
  authorAvatar: string,
  rating: number,
  opinion: string,
  publishDate: string,
}

export interface ReviewFormData {
  comment: string, 
  rating: number, 
  apartmentId: number,
}

  
