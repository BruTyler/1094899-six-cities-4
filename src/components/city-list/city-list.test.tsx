import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CityList from './city-list';
import {City} from '../../types';

const EMPTY_HANDLER = () => null;
const CITIES: City[] = [
  {
    id: `A`,
    location: [1, 2],
    defaultZoom: 12
  },
  {
    id: `C`,
    location: [2, 3],
    defaultZoom: 12
  },
];

describe(`<CityList /> render suit`, () => {
  it(`<CityList /> render city list`, () => {
    const generatedTree = renderer.create(
        <CityList
          cityList={CITIES}
          activeCity={CITIES[0]}
          onCitySelect={EMPTY_HANDLER}
        />
    ).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});

