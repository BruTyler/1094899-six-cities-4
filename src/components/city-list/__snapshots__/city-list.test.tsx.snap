// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityList /> render suit <CityList /> render city list 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
        onClick={[Function]}
      >
        <span>
          A
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          C
        </span>
      </a>
    </li>
  </ul>
</section>
`;
